CareKit/CareCard/OCKCareCardTableViewCell.m

```diff
@@ -109,13 +109,13 @@ - (void)prepareView {
                   forControlEvents:UIControlEventTouchUpInside];
         
         if (self.delegate &&
```

Comment:
Here we're checking if the delegate is set and if it responds to a selector, but to the best of my knowledge there are presently no cases where the delegate is not set or the method in question is not implemented.

The two places this cell is used are `OCKCareCardViewController` and `OCKCareContentsViewController`, and in both cases the cell delegate is set in the cellForIndexPath table view data source methods. Moreover, as of this PR both have a default implementation of the selectors in question that just returns `NULL`. 

Do you think it's better to leave these two checks in place incase `OCKCareCardButton` ever becomes a public class, or should we take out the checks since they are superfluous?

Reply:
@moppymopperson, I agree, it is always a good idea to check if delegate exists, and if delegate responds to the selector, before calling it.

Reply:
@akshay-yadav-apple Glad to hear that, thanks for the feedback.